### PR TITLE
Implement OneOf. Refactor compose classes. 

### PR DESCRIPTION
**Pull request by iver56**
Also add `apply_to_children` argument in some methods in BaseCompose. It defaults to `True`, so the change is backwards compatible.

#61

### CODE REVIEW

File: audiomentations/core/composition.py [hunk 1]
```diff
@@ -19,59 +20,134 @@ def randomize_parameters(self, *args, **kwargs):
...
             if self.shuffle:
                 random.shuffle(transforms)
             for transform in transforms:
                 samples = transform(samples, sample_rate)
 
         return samples
 
-    def randomize_parameters(self, samples, sample_rate):
-        for transform in self.transforms:
-            transform.randomize_parameters(samples, sample_rate)
```
Comment (Cangonin) [inline audiomentations/core/composition.py:58]: Why did you remove it ?

File: audiomentations/core/composition.py [hunk 2]
```diff
@@ -19,59 +20,134 @@ def randomize_parameters(self, *args, **kwargs):
...
+
+    def randomize_parameters(self, *args, **kwargs):
+        super().randomize_parameters(*args, **kwargs)
+        self.should_apply = random.random() < self.p
+        if self.should_apply:
+            self.transform_index = random.randint(0, len(self.transforms) - 1)
+
+    def __call__(self, *args, **kwargs):
+        if not self.are_parameters_frozen:
+            self.randomize_parameters(apply_to_children=False)
```
Comment (Cangonin) [inline audiomentations/core/composition.py:143]: So when `apply_to_children` is set to False, what does it mean here ? That you randomize parameterst to the only transform which will be applied ?

**Review by Cangonin — APPROVED** [review]

File: audiomentations/core/composition.py (hunk 2 above)
Comment (iver56) [reply audiomentations/core/composition.py:143]: It means that it only randomize the parameters (self.should_apply and self.transform_index) of "self", and not the parameters in the list self.transforms. The parameters in each transform will be randomized when it gets called.

File: audiomentations/core/composition.py (hunk 1 above)
Comment (iver56) [reply audiomentations/core/composition.py:58]: Because I implemented it in the base class instead :)
